lib/ContextModuleFactory.js

```diff
@@ -111,12 +114,17 @@ module.exports = class ContextModuleFactory extends ModuleFactory {
 
 				const contextResolver = this.resolverFactory.get(
 					"context",
-					resolveOptions || EMPTY_RESOLVE_OPTIONS
-				);
-				const loaderResolver = this.resolverFactory.get(
-					"loader",
-					EMPTY_RESOLVE_OPTIONS
+					resolveOptions
+						? dependencies.length > 0
+							? cachedSetProperty(
+									resolveOptions,
+									"dependencyType",
+									dependencies[0].category
+							  )
+							: resolveOptions
+						: undefined
```

Comment:
```suggestion
					dependencies.length > 0
						? cachedSetProperty(
								resolveOptions || EMPTY_RESOLVE_OPTIONS,
								"dependencyType",
								dependencies[0].category
						  )
						: resolveOptions
```

Even if there are no resolveOptions, we should set the dependencyType.

Reply:
thanks, missed that..